### PR TITLE
feature : add Record and Achievements

### PR DESCRIPTION
**Pull request by fkrdud1125**
- Changes in AchievementManager : Total Score, Total Playtime, Accuracy, Perfect Stage, Flawless Failure(if died without hitting anything / Achievement for fun), Best Friends(for 2-player mode)
- Changes in DrawManager : Add multiple function which draws the Achievement Menu
- Changes in FileManager : Add functions that loads and saves Total Score, Total Playtime, and 4 Achievements
- Name Limit : change to 6 -> 4 because the width of achievement screen
- Changes in GameState : Add getAccuracy(fixed Zero Division problem)
- Changes in AchievementScreen : Add functions to manage the drawing of Achievement
___________________________________________________________________________________________________________
- Collaborations with Coin team : Coins earned when completing each achievement are stored in the deposit of Wallet

### CODE REVIEW

**Review by PIut0 — APPROVED** [review]
LGTM :)

**Comment by idealtrue0** [comment]
Previously, the total number of stages was up to level 7, but our team created endless levels. Is it okay to only give achievements up to level 7?

기존에는 총 스테이지가 7단계까지였으나, 저희 팀에서 그뿐만이 아닌 끝없는 단계를 만들었는데, 업적을 7단계까지만 해도 괜찮은가요?

File: src/screen/ScoreScreen.java [hunk 1]
```diff
@@ -78,17 +80,12 @@ public ScoreScreen(final int width, final int height, final int fps,
 		this.coinsEarned = (int)Math.round(gameState.getScore() * coin_ratio);
 
 		// deposit the earned coins to wallet
+		this.accuracy = gameState.getAccuracy();
+		this.coinsEarned = gameState.getScore()/10 + achievementManager.getAchievementReward();
```
Comment (bumsoft) [inline src/screen/ScoreScreen.java]: gameState.getScore()/10 부분은 80번째 줄((int)Math.round(gameState.getScore() * coin_ratio);)로 변경되었습니다!
this.coinsEarned += achievementManager.getAchievementReward(); 로 변경만 부탁드립니다!

**Review by bumsoft — CHANGES_REQUESTED** [review]
coinsEarned 부분 수정 및 src/screen/ScoreScreen.java만 충돌해결 부탁드립니다!
Please correct the coinsEarned part and resolve the src/screen/ScoreScreen.java conflict!

**Review by bumsoft — APPROVED** [review]

**Pull request merged**